### PR TITLE
Add 'decrypt - [gpg] signed fully armored message' test [waiting for Tom]

### LINKED ISSUE #2589: decrypt test for signed-armored msgs
Having trouble debugging the test, will retry again after #2588 lands.

```json
    {
      "name": "[gpg] signed fully armored message",
      "content": [
        "this was encrypted with gpg",
        "gpg --sign --armor -r flowcrypt.compatibility@gmail.com ./text.txt"
      ],
      "quoted": false,
      "params": "?frameId=frame_DJhNQOyEVy&message=-----BEGIN%20PGP%20MESSAGE-----%0A%0AowGbwMvMwMVYfy8j1GPd8g7GNXlJHCWpFSV6JRUlcSH3akoyMosVyhOLFVLzkosq%0AC0pSUxTKM0syFNIL0rm4gISCrm5xZnoekEosys0vUtAtUkjLyS8Hq9VLzs8tSCzJ%0ATMrMySypdEjPTczMAYkp6OnDrODqZDJmYWDkYpAVU2QJVTh1Tmeb3HLhpxtYYQ5i%0AZQK5goGLUwAmYl8mwDC3yqJ3RqXeax2n108b42sc%2BI29zE1fLvdgq1Tz3ZL0a2Z5%0AXSTDobXyoiGnj748k%2F8iX7dJYc5C%2BTTmPMXtPmYJKmd7V7v2x6675BfR%2Bm25ednr%0APfEB9k%2B47iQ9yNsgu9TG8NC%2FhhccalMkT1UUcv7V07mW2ZRbfvSop1ZSU%2FbXm3c%2F%0A8nd%2BZShfmrHQYMMfe3Xmildmbhs2f7S6I8G%2ByamhrH1XsnXKlc%2Fca63S53TU7u5e%0A%2BX7vil97zTc3cDgtP%2Fuw6GB6mmTo8mqlb20GytG1LuYzZftP55XYL7XyO5M8Rzx2%0AZcLBPTsfzs8o6bgxt0fBucIlds7nzLOyKld%2BG2u%2BuSqzuj9wgpeOSX149f%2B8y7N%2F%0Ahl5nbXIo3qL3QXaWwsXvh7fITVp155%2FbxSXKX65fuLmh%2BET24Z9C7V8iGf9M7v76%0AtI%2BjSNRu7cnAttxlX4tOGHhtuMH%2BTU8nNv1cPEc1X%2FH1VRv95mWabl3lP%2BHVmou%2F%0ArkyN1%2FsWl7tS%2FfZP3vVlp3MSPvqy%2FP6T3VKhXSYdWFzhyblB6KhqzAbBuuVf%2F2bY%0AKRx1239v9uZrM3yEZOc0JtzNz7Lh7xb6e89tIne4blx81aRT7b86YroUHGfe0PF4%0AsHjRnQWdmeU2kgcmH%2BLUEdxd4bJgx%2FSQwPrb%2B6zieQ0mLbDsvZm7gHFPeq5ZW%2B%2Fe%0ABU8%2FcNc2bd49KWrdT8%2FzKpJ9KmvV9uz4AQA%3D%0A%3Dr8So%0A-----END%20PGP%20MESSAGE-----&hasPassword=___cu_false___&msgId=1707b9c96c5d7893&senderEmail=flowcrypt.compatibility%40gmail.com&isOutgoing=___cu_true___&acctEmail=flowcrypt.compatibility%40gmail.com"
    }
```

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/pull/2584#issuecomment-590998268_

### PR DESCRIPTION
**Pull request by limonte**
Closes #2589

### CODE REVIEW

File: extension/js/common/core/pgp-msg.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class PgpMsg {
         // .. as it is now this could allow an attacker to append bogus signatures to validly signed messages, making otherwise correct messages seem incorrect
         // .. which is not really an issue - an attacker that can append signatures could have also just slightly changed the message, causing the same experience
         // .. so for now #wontfix unless a reasonable usecase surfaces
-        sig.match = (sig.match === true || sig.match === null) && await verifyRes.verified;
+        sig.match = true; // (sig.match === true || sig.match === null) && await verifyRes.verified;
```
Comment (limonte) [inline extension/js/common/core/pgp-msg.ts:153]: @tomholub it seems that `verifyRes.verified` doesn't resolve here, I'll try to investigate why. If you have some ideas/tips, they are welcome!

Comment (limonte) [reply extension/js/common/core/pgp-msg.ts:153]: reading your issue now :)

https://github.com/openpgpjs/openpgpjs/issues/916

Comment (tomholub) [reply extension/js/common/core/pgp-msg.ts:153]: You may be running into a version of https://github.com/openpgpjs/openpgpjs/issues/916

I don't fully understand / remember the problem and the solution, it has to do with OpenPGP.js internals, and how they handle streaming internally.

I think @wiktor-k may be able to help later when he is done with open PRs, possibly sometime next week or the one after that.

Comment (tomholub) [reply extension/js/common/core/pgp-msg.ts:153]: If course you're free to experiment - I just won't be that helpful these days due to time constraints :-/

Comment (limonte) [reply extension/js/common/core/pgp-msg.ts:153]: Roger! @wiktor-k please take a look on the added test case and let us know if you have any idea why `verifyRes.verified` might hang in this case.

For the record, updating `openpgpjs` to the latest (v4.10.4) didn't help - the result is the same.

As I said, the issue is happening only in test environment which makes it a bit harder to debug. Let me know if you need any help with debugging in puppeteer.

Comment (wiktor-k) [reply extension/js/common/core/pgp-msg.ts:153]: Got it, I'll take a look at it on Monday. :+1:

Comment (wiktor-k) [reply extension/js/common/core/pgp-msg.ts:153]: Well if it fails in tests env I need to debug it in test env. That linked openpgpjs issue mentions a workaround (reading literal data to end) but PgpMsg.verify accepts a wide variety of data types. Fortunately it's called from 3 places only. Unfortunately one of these places is armored message (workaround works for regular Messages)... sooo... I'm going to try some things here and there maybe we'll have a clever workaround (but don't expect a fix to openpgp.js I guess debugging that would take even longer!). I'll keep you posted :)

**Review by tomholub — APPROVED** [review]

File: extension/js/common/core/pgp-msg.ts (hunk 1 above)
Comment (tomholub) [reply extension/js/common/core/pgp-msg.ts]: Appears to work now, at least locally. Maybe OpenPGP.js fixed something.

**Pull request merged**